src/android/Mapbox.java

```diff
@@ -168,7 +183,23 @@ public void run() {
             FrameLayout.LayoutParams params = new FrameLayout.LayoutParams(webViewWidth - left - right, webViewHeight - top - bottom);
             params.setMargins(left, top, right, bottom);
             mapView.setLayoutParams(params);
-
+			//@anothar we have to override back button or app will just close
+            mapView.setOnKeyListener(new View.OnKeyListener() {
```

Comment:
It's more like cordova plugin development feature. Cordova subscribes to Android webview events. But plugin adds new control in front of webview. Map control is not the child of webview. So webview knows nothing about keys pressed inside map. And thus it can't delegate events to cordova. We have to do that manually.